File: NumberGuesser/guess.js
"user strict";

// Générer le nombre que le joueur devra déviner
const to_guess = Number.parseInt(Math.random()*10)

// Le nombre total d'essais qu'a l'utilisateur
let total_rounds = 10

while (total_rounds > 0) {
	// Démander à l'utilisateur une valeur
	let user_guess = Number.parseInt(prompt("Dévinez la valeur : "))

	// Decrémenter la valeur de total_rounds
	total_rounds--

	// Si les deux valeurs correspondent
	if (total_rounds ==  to_guess) {

		// Il a gagné-On quitte la boucle

		const msg = "Vous avez gagné !"
		// Créer un élément HTML

		const p = document.createElement('p')

		// Lui ajouter ce texte
		p.innerText = msg;

		// Ajouter cet élément au flux de la page
		document.querySelector('d').insertAdjacentHTML("beforeend",p);

		// Afficher aussi dans la console
		console.log(msg)

		// Rompre la boucle
		break
	} else {

		// Le message 
		const msg = "Vous avez perdu ! Tour " + total_rounds

		// Créer un élément HTML
		const p = document.createElement('p')

		// Lui ajouter ce texte
		p.innerText = msg;

		// Ajouter cet élément au flux de la page
		document.querySelector('.d').insertAdjacentHTML("beforeend",p);


		// Afficher aussi dans la console
		console.log(msg)

		// Continuer de redémander des valeurs
		continue
	}
}

File: NumberGuesser/guess.py
# coding : utf-8

import random as r 

# Générer la valeur que le joueur devra déviner
to_guess = round(r.random()*10)

# Le nombre total d'essais qu'a l'utilisateur
total_rounds = 10

while total_rounds > 0:
	# Demander à l'utilisateur une valeur
	user_guess = int(input("Dévinez la valeur : "))

	# Decrémenter la valeur total_rounds
	total_rounds-=1

	# Si les deux valeurs sont les mêmes 
	if int(to_guess) == int(user_guess):
		# Il a gagné-On quitte la boucle
		print('Vous avez gagné !')
		break
	# Sinon
	else:
		# Afficher un message - Redémander une valeur
		print('Echec ! Tour ' + str(total_rounds))
		continue